### PR TITLE
All non-test Typescript files have < 600 lines of Code 

### LINKED ISSUE #1298: Ensure all non-test Typescript files have < 600 lines - Admin
**Is your feature request related to a problem? Please describe.**

Our  GitHub action `countline.py` script is not sufficiently effective as deployed

- https://github.com/PalisadoesFoundation/talawa-admin/tree/main/.github/workflows

We need to also restore our goal of having no typescript files with > 600 lines of code. We do this for readability and to force better coding standards by encouraging the use of more reusable code.

**Describe the solution you'd like**

1. Update our pull request GitHub action to make the `countline.py` script fail for files with > 600 lines of code. Run the  `./countline.py --help` command to see how that is done.
1. Run  the `countline.py` script with the 600 line limit to identify the offending files.
1. Refactor the code in the offending files so that the script does not fail.
**Describe alternatives you've considered**

**Approach to be followed (optional)**
- See above

**Additional context**
- Please work with @noman2002 in reviewing your PR

**Potential internship candidates**
- Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Sauradip07**
**What kind of change does this PR introduce?**
Feature

**Issue Number:** Ensure all non-test Typescript files have < 600 lines - Admin

Fixes #1298 

**Did you add tests for your changes?**

 No

**Snapshots/Videos:**

Screenshot from 2024-01-09 22-04-58

**Summary**

The modified code checks TypeScript files (*.ts and *.tsx) excluding test files, ensuring they do not exceed a specified line limit. It iterates through the directory, printing paths of files exceeding the limit, and returns True if all files comply, and False otherwise. Adjust file extensions as needed.

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by Sauradip07** [comment]
The GitHub workflow is designed to check for unauthorized file changes, as I am working on `countline.py` Checking for unauthorized file changes the workflow fails.

Additionally, the workflow successfully detected the change. It pointed out that the file` ‘./src/GraphQl/Queries/Queries.ts’ `contains more than 600 lines of code, specifically 863 lines. With proper file path that makes developer life easy to debug it.

This is a significant amount of code, and it’s important to manage such large files effectively to maintain code readability and manageability.
Please feel free to reach out if there are any modifications or any changes need to be made.

File: .github/workflows/countline.py [hunk 1]
```diff
@@ -193,6 +193,31 @@ def _arg_parser_resolver():
...
+        Args:
+            limit: Maximum allowable lines of code in a file.
+            dir_path: Path of the directory to check.
+
+        Returns:
+            result: Returns True if all non-test TypeScript files contain fewer 
+                    or equal to 'limit' lines of code. Returns False if any file 
+                    contains more than 'limit' lines of code.
+    """
+    for root, dirs, files in os.walk(dir_path):
```
Comment (palisadoes) [inline .github/workflows/countline.py:209]: Nothing needed to be added to the script it works correctly

File: .github/workflows/countline.py [hunk 2]
```diff
@@ -277,6 +302,15 @@ def main():
 
             print("  Line count: {:>5} File: {}".format(line_count, filepath))
 
+    # Check TypeScript file length
+    dir_path = '.'  # directory path
+    limit = 600  # line limit
+    if not check_file_length(limit, dir_path):
```
Comment (palisadoes) [inline .github/workflows/countline.py:308]: Nothing needed to be added to the script it works correctly

**Review by palisadoes — COMMENTED** [review]
1. All that needed to be done for this issue was to look at this GitHub action file:
    1. https://github.com/PalisadoesFoundation/talawa-admin/blob/develop/.github/workflows/pull-requests.yml
1. Then find this section specifying 1000 lines
    image
1. Run the countline.py script with a limit of 600 lines with all the other parameters being the same.
1. Verify that no files are over 600 lines in length (the script already counts lines)
1. Refactor any files that are over 600 lines 
1. Reset https://github.com/PalisadoesFoundation/talawa-admin/blob/develop/.github/workflows/pull-requests.yml to use 600 lines instead for the script

Please follow these steps

**Comment by palisadoes** [comment]
1. Closing the PR. You have made no changes to any typescript files. 
2. I'll also close the issue as it appears that we are compliant

**Pull request closed without merging**